### PR TITLE
Add `POWER`, `EXP`, `LOG`, `LOG10`, and `LN` functions

### PR DESCRIPTION
**Pull request by HactarCE**

### CODE REVIEW

File: quadratic-core/src/formulas/functions/mathematics.rs [hunk 1]
```diff
@@ -242,6 +242,67 @@ fn get_functions() -> Vec<FormulaFunction> {
...
+                } else {
+                    return Err(RunErrorMsg::NaN.with_span(span));
+                }
+            }
+        ),
+        formula_fn!(
+            /// Returns the [base-10 logarithm] of `number`.
+            ///
+            /// [base-10 logarithm]:
+            ///     https://en.wikipedia.org/wiki/Common_logarithm
```
Comment (ddimaria) [inline quadratic-core/src/formulas/functions/mathematics.rs:288]: Is this the correct link?

File: quadratic-core/src/formulas/functions/mathematics.rs [hunk 2]
```diff
@@ -614,6 +675,72 @@ mod tests {
...
+            }
+        }
+    }
+
+    #[test]
+    #[parallel]
+    fn test_pow_0_0() {
+        // See https://en.wikipedia.org/wiki/Zero_to_the_power_of_zero
+
+        // This is compatible with Google Sheets, but not Excel
```
Comment (ddimaria) [inline quadratic-core/src/formulas/functions/mathematics.rs:703]: @davidkircos This may present an error case when importing functions from an Excel file.

**Review by ddimaria — APPROVED** [review]
Looks great, one possible hyperlink issue.

File: quadratic-core/src/formulas/functions/mathematics.rs (hunk 1 above)
Comment (HactarCE) [reply quadratic-core/src/formulas/functions/mathematics.rs:288]: Yep, it's the correct link

File: quadratic-core/src/formulas/functions/mathematics.rs [hunk 3]
```diff
@@ -614,6 +675,72 @@ mod tests {
...
+    }
+
+    #[test]
+    #[parallel]
+    fn test_negative_pow() {
+        let g = Grid::new();
+        let e = RunErrorMsg::NaN;
+
+        assert_eq!("1", eval_to_string(&g, "POWER(-2, 0)"));
+        assert_eq!("1", eval_to_string(&g, "(-2)^0"));
```
[the remark this answers is not in the document]
Comment (HactarCE) [reply quadratic-core/src/formulas/functions/mathematics.rs:732]: Yep, Excel does too

**Review by davidkircos — APPROVED** [review]
All these functions seem to work great!

Nice!

**Pull request merged**